## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-fs.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-erasure.sh | :heavy_check_mark: |
| mint-dist-erasure.sh | :heavy_check_mark: |
| mint-compress-encrypt-dist-erasure.sh | :heavy_check_mark: |
| mint-pools.sh | :heavy_check_mark: |
| mint-large-bucket.sh | :x: [more...](#4e1ad903-1ef4-43bf-8711-68245d7c841a) |
<a name="4e1ad903-1ef4-43bf-8711-68245d7c841a"></a>
### 15709-8d4d3e9/mint-large-bucket.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-k8s7:32028
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0
RUN_ON_FAIL:          0

To get logs, run 'docker cp :/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... done in 1 seconds
(2/14) Running aws-sdk-java tests ... done in 1 seconds
(3/14) Running aws-sdk-php tests ... done in 44 seconds
(4/14) Running aws-sdk-ruby tests ... done in 5 seconds
(5/14) Running awscli tests ... done in 58 seconds
(6/14) Running healthcheck tests ... done in 0 seconds
(7/14) Running mc tests ... done in 30 seconds
(8/14) Running minio-go tests ... done in 1 minutes and 10 seconds
(9/14) Running minio-java tests ... FAILED in 1 minutes and 21 seconds
Exception in thread "main" java.lang.AssertionError: object count; expected=1050, got=1049 expected:<1049> but was:<1050>
	at org.junit.Assert.fail(Assert.java:88)
	at org.junit.Assert.failNotEquals(Assert.java:834)
	at org.junit.Assert.assertEquals(Assert.java:645)
	at FunctionalTest.testListObjects(FunctionalTest.java:1412)
	at FunctionalTest.listObjects(FunctionalTest.java:1457)
	at FunctionalTest.runObjectTests(FunctionalTest.java:3743)
	at FunctionalTest.runTests(FunctionalTest.java:3781)
	at FunctionalTest.runEndpointTests(FunctionalTest.java:3881)
	at FunctionalTest.main(FunctionalTest.java:4002)
(9/14) Running minio-js tests ... done in 1 minutes and 4 seconds
(10/14) Running minio-py tests ... done in 1 minutes and 25 seconds
(11/14) Running s3cmd tests ... done in 25 seconds
(12/14) Running s3select tests ... done in 4 seconds
(13/14) Running versioning tests ... done in 3 minutes and 5 seconds

Executed 13 out of 14 tests successfully.
```
Deleting image on docker hub
Deleting image locally

--- a/cmd/main.go
+++ b/cmd/main.go
@@ -23,11 +23,13 @@ import (
 	"os"
 	"path/filepath"
 	"runtime"
+	"runtime/debug"
 	"sort"
 	"strings"
 
 	"github.com/minio/cli"
 	"github.com/minio/minio/internal/color"
+	"github.com/minio/minio/internal/logger"
 	"github.com/minio/pkg/console"
 	"github.com/minio/pkg/trie"
 	"github.com/minio/pkg/words"
@@ -189,8 +191,27 @@ func Main(args []string) {
 	// Set the minio app name.
 	appName := filepath.Base(args[0])
 
+	if os.Getenv("_MINIO_DEBUG_NO_EXIT") != "" {
+		freeze := func(_ int) {
+			// Infinite blocking op
+			<-make(chan struct{})
+		}
+
+		// Override the logger os.Exit()
+		logger.ExitFunc = freeze
+
+		defer func() {
+			if err := recover(); err != nil {
+				fmt.Println("panic:", err)
+				fmt.Println("")
+				fmt.Println(string(debug.Stack()))
+			}
+			freeze(-1)
+		}()
+	}
+
 	// Run the app - exit on error.
 	if err := newApp(appName).Run(args); err != nil {
-		os.Exit(1)
+		os.Exit(1) //nolint:gocritic
 	}
 }

--- a/internal/logger/console.go
+++ b/internal/logger/console.go
@@ -32,6 +32,9 @@ import (
 // ConsoleLoggerTgt is a stringified value to represent console logging
 const ConsoleLoggerTgt = "console+http"
 
+// ExitFunc is called by Fatal() class functions, by default it calls os.Exit()
+var ExitFunc = os.Exit
+
 // Logger interface describes the methods that need to be implemented to satisfy the interface requirements.
 type Logger interface {
 	json(msg string, args ...interface{})
@@ -90,7 +93,7 @@ func (f fatalMsg) json(msg string, args ...interface{}) {
 	}
 	fmt.Println(string(logJSON))
 
-	os.Exit(1)
+	ExitFunc(1)
 }
 
 func (f fatalMsg) quiet(msg string, args ...interface{}) {
@@ -143,7 +146,7 @@ func (f fatalMsg) pretty(msg string, args ...interface{}) {
 	}
 
 	// Exit because this is a fatal error message
-	os.Exit(1)
+	ExitFunc(1)
 }
 
 type infoMsg struct{}

